core/storage/mpool/mpool.cpp

```diff
@@ -20,6 +26,424 @@ namespace fc::storage::mpool {
   using primitives::tipset::HeadChangeType;
   using vm::message::UnsignedMessage;
 
+  constexpr GasAmount kMinGas{1298450};
+  constexpr size_t kMaxBlocks{15};
+  constexpr size_t kMaxBlockMessages{16000};
+
+  TokenAmount getGasReward(const UnsignedMessage &msg,
+                           const TokenAmount &base_fee) {
+    TokenAmount reward{
+        std::min<TokenAmount>(msg.gas_fee_cap - base_fee, msg.gas_premium)
+        * msg.gas_limit};
+    if (reward < 0) {
+      // https://github.com/filecoin-project/lotus/blob/191a05da4872bf9849f178e6db5c0d6e87d05baa/chain/messagepool/selection.go#L687
+      reward *= 3;
+    }
+    return reward;
+  }
+
+  double getGasPerf(const TokenAmount &reward, GasAmount limit) {
+    return (double)boost::multiprecision::cpp_rational{reward * kBlockGasLimit,
+                                                       limit};
+  }
+
+  auto blockProbabilities(double ticket_quality) {
+    // TODO(turuslan): what to use from boost/math/distributions/poisson.hpp
+    static auto no_winners{[] {
+      std::vector<double> out;
+      out.resize(kMaxBlocks);
+      auto mu{5};
+      auto cond{log(-1 + exp(mu))};
```

Comment:
Please rename variables for something more informative than mu, cond etc.

Reply:
I think mu is for μ.